ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -123,7 +123,9 @@ export default class SearchSelectWithModal extends Component {
   @action
   handleChange() {
     if (this.selectedOptions.length && typeof this.selectedOptions[0] === 'object') {
-      this.args.onChange(Array.from(this.selectedOptions, (option) => option.id));
+      const onChangeValue = Array.from(this.selectedOptions, (option) => option.id);
+      // if returnString is true, the api expects the value to be a string (ex: identityTokenKey) otherwise set the value to an array of strings.
+      this.args.returnString ? this.args.onChange(onChangeValue[0]) : this.args.onChange(onChangeValue);
```

Comment:
this logic feels like something that should happen in the form component or serializer 🤔 

it's hard to think of suggestions since the modal functionality is coming in the follow on PR, can this logic be removed for now and reconsidered when the functionality is implemented?